tests/installation/installation_overview.pm

```diff
@@ -69,6 +69,18 @@ sub run {
         $self->deal_with_dependency_issues;
         assert_screen "inst-xen-pattern" if get_var('XEN');
         ensure_ssh_unblocked;
+        # Check the systemd target, see poo#45020
+        if (get_var('DESKTOP')) {
+            # QAM do some installs with DESKTOP=textmode, but install gnome... 
+            # and switching between ttys does not work for hyperv.
+            return if (get_var(QAM_MINIMAL) || check_var('MACHINE', 'svirt-hyperv'));
+            my $target = check_var('DESKTOP', 'textmode') ? "multi-user" : "graphical";
+            select_console 'install-shell';
+            # The default.target is not yet linked, so we have to parse the logs.
+            assert_script_run("grep 'target has been set' /var/log/YaST2/y2log |tail -1 |grep \"$target\"", fail_message => "Wrong systemd target detected, aborting");
```

Comment:
Perhaps it make sense to pick a few more lines (more than 1) in case we could not see the result, WDYT?

Reply:
One grep should be enough, you can use grep -E for regex but probably in this case you don't even need it. Move $target to first grep.

Reply:
In some cases, we have patterns add a desktop environment to a minimal install. So we need two greps here, because if we had an earlier line (before patterns) matching what we expect, it is a false postitive. So we want all lines with 'target has been set' and then keep only the last one.

Reply:
I understand, thanks, I was thinking in the other way around, to check with `tail` the last lines written in the logs to avoid the false positive and do one grep, but maybe for your case is not the best way to go.

Reply:
The y2logs are extremely verbose, so grep is much safer than tail in this case.